packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -288,110 +288,141 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
       ? rootProps.showCellRightBorder
       : !removeLastBorderRight && rootProps.disableExtendRowFullWidth;
 
-    const cellParams = apiRef.current.getCellParams(rowId, column.field);
+    if (row) {
```

Comment:
Could move the content of this `if` to a function, e.g. `getCell`? Each branch is very long to read.